WordPress/gutenberg
# Pages data view: Add Pending and Private views

Follow-up to https://github.com/WordPress/gutenberg/pull/62021. I'm unsure why these weren't included in https://github.com/WordPress/gutenberg/issues/59659. I figured we can discuss in a PR.

## What?
Adds views for pending and private pages in the Pages data view.

## Why?
There are views for each of the other statuses (published, drafts, scheduled, trashed). It seems reasonable to include these too, particularly as there are such views in the wp-admin pages table.

<img width="358" alt="Screenshot 2024-05-30 at 12 56 21" src="https://github.com/WordPress/gutenberg/assets/846565/5b21ae75-f08d-42f7-b94c-7855da718c4f">

I think there's room to explore alternative icons in follow-ups during beta.


## Testing Instructions
1. Open the Pages data view
2. Observe the new 'Private' and 'Pending' views
3. Ensure they work as expected
